tests/ai_hub/utils.py

```
@@ -1154,3 +1156,35 @@ def execute_authenticated_post(url: str, token: str, files: dict[str, tuple[str,
+    if ready_replicas >= desired_replicas and updated_replicas >= desired_replicas:
+        LOGGER.info(f"OAuth deployment ready: ready={ready_replicas}/{desired_replicas}")
+        return True
+    raise OAuthDeploymentNotReady(
+        f"OAuth not ready: ready={ready_replicas}/{desired_replicas}, updated={updated_replicas}"
+    )
+
+
+def _restart_oauth_and_wait() -> None:
+    """Rolling restart of OAuth deployment to pick up IDP config change, then wait for readiness."""
+    LOGGER.info("Rolling restart of OAuth deployment to pick up IDP config change")
+    run_oc(  # noqa: FCN001
```

Ruff (0.16.0)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)